src/c++/perf_analyzer/genai-perf/README.md

```diff
@@ -1,30 +1,260 @@
+
+genai-perf --help
 ```
 
-## Examples
+# Basic Usage
+
+## Triton with TRT-LLM
 
+```bash
+genai-perf -m llama-2-7b --concurrency 1 --service-kind triton --output-format trtllm
 ```
-# Profile an LLM with hard-coded inputs
-genai-perf -m my_llm_model
+
+## Triton with vLLM
+
+```bash
+genai-perf -m llama-2-7b --concurrency 1 --service-kind triton --output-format vllm
 ```
 
-## Test
+## OpenAI Chat Completions Compatible APIs
+
+https://platform.openai.com/docs/api-reference/chat
 
+```bash
+genai-perf -m llama-2-7b --concurrency 1 --service-kind openai --endpoint v1/chat/completions --output-format openai_chat_completions
 ```
-pip install .
-pytest tests/
+
+## OpenAI Completions Compatible APIs
+
+https://platform.openai.com/docs/api-reference/completions
+
+```bash
+genai-perf -m llama-2-7b --concurrency 1 --service-kind openai --endpoint v1/completions --output-format openai_completions
 ```
+
+# Model Inputs
+GenAI-Perf supports model inputs from either the HuggingFace OpenOrca or
+CNN_DailyMail datasets or it can create synthetic input data. This is specified
+using the `--input-type` CLI option.
+
+When the dataset is coming from HuggingFace you can specify the following
+options:
+* `--dataset`: HuggingFace dataset to use for benchmarking.
+
+When the dataset is synthetic you can specify the following options:
+* `--num-of-output-prompts`: The number of synthetic output prompts to generate
+* `--input-tokens-mean`: The mean number of tokens of synthetic input data.
+* `--input-tokens-stddev`: The standard deviation number of tokens of synthetic
+  input data.
+* `--random-seed`: The seed used to generate random values.
+
+# Metrics
+
+GenAI-Perf collects a diverse set of metrics that captures the performance of
+the inference server.
+
+| Metric | Description | Aggregations |
+| - | - | - |
+| Time to First Token | Time between when a request is sent and when its first response is received, one value per request in benchmark | Avg, min, max, p99, p90, p75 |
+| Inter Token Latency | Time between intermediate responses for a single request divided by the number of generated tokens of the latter response, one value per response per request in benchmark | Avg, min, max, p99, p90, p75 |
+| Request Latency | Time between when a request is sent and when its final response is received, one value per request in benchmark | Avg, min, max, p99, p90, p75 |
+| Number of Output Tokens | Total number of output tokens of a request, one value per request in benchmark | Avg, min, max, p99, p90, p75 |
+| Output Token Throughput | Total number of output tokens from benchmark divided by benchmark duration | None–one value per benchmark |
+| Request Throughput | Number of final responses from benchmark divided by the benchmark duration | None–one value per benchmark |
+
+# CLI
+
+###### `-h`
+###### `--help`
+
+###### `-v`
```

Comment:
For commands with a short and long option, should we put them on the same line?

Reply:
Perhaps separated by a comma?

Reply:
I was just copying git documentation: https://git-scm.com/docs/git

I can't think of any convincing arguments for one way or the other. All on same line might be less readable? Especially when there are a list of fixed values that can be put:

```
`-a {fixed_value_1, fixed_value_2, fixed_value_3}, --my-long-option-a {fixed_value_1, fixed_value_2, fixed_value_3}
```

Also when they're combined, the anchor link becomes slightly less readable.

Reply:
seems like a valid reason to leave them alone for now.
Thanks for the perspective.